scripts/ci-checks.sh

```diff
@@ -50,7 +52,7 @@ fi
 # Virtual Environment w/ dependencies for Python steps
 if [ ! -f "scripts/env/bin/activate" ]
     then
-        python3.8 -m venv scripts/env
+        $PYTHON -m venv scripts/env
 fi
 
 source scripts/env/bin/activate
```

Comment:
After this line, `python` will refer to the one from the venv, so I don't really understand why the fix below would make a difference.

Reply:
That's a good point. It does so anyways; mine used Python 3.6 which doesn't have the `dirs_exist_ok` argument on `shutil.copytree`.

Reply:
And it still does so if I change line 73 back to using `python`: `191,30,error,unexpected-keyword-arg:Unexpected keyword argument 'dirs_exist_ok' in function call`

Reply:
When you run `source scripts/env/bin/activate` manually, what do the following print?
```
python -c "import sys;print(sys.executable)"
python3.8 -c "import sys;print(sys.executable)"
```

Reply:
```
...$ . scripts/env/bin/activate
(env) ...$ python -c "import sys;print(sys.executable)"
/usr/bin/python
(env) ...$ python3.8 -c "import sys;print(sys.executable)"
/usr/bin/python3.8
(env) ...$ ls -la /usr/bin/python
lrwxrwxrwx 1 root root 24 Mar  8 12:36 /usr/bin/python -> /etc/alternatives/python
(env) ...$ ls -la /etc/alternatives/python
lrwxrwxrwx 1 root root 18 Mar  8 12:36 /etc/alternatives/python -> /usr/bin/python3.6
```

Reply:
Weird, something is very broken on your system, let's figure this out "offline".

Reply:
It's the venv; for some reason I had a root-owned `scripts/env` and never got any error messages relating to that. Now that I tried to manually recreate it, I got the permission error and I was able to create a new venv that works as expected. 

But yes, my system is broken, because I tried to get rid of Python 2.x while too much stuff still depends on it.

Still a good idea to have a `$PYTHON` I think.

Reply:
Glad you figured it out. `$PYTHON` makes only sense before venv activation, otherwise you would also have to patch up related calls to `pip` or Python entrypoints like `mypy`. I think whenever we update to a newer Python version we search-replace all instances of `pythonx.y`, so it wouldn't really be less maintenance I think.

Reply:
Yeah, that's right, all the `pip` references would have to change too. Is `python` inside of a venv guaranteed (or at least intended) to be the same version of python that created the venv?

Reply:
Yes, that's guaranteed.